packages/web3-providers-ws/src/types.ts

```diff
@@ -0,0 +1,29 @@
+import { OutgoingHttpHeaders } from 'http';
+import { JsonRpcPayload, JsonRpcResponse } from 'web3-common';
+import { IClientConfig } from 'websocket';
+import { DeferredPromise } from './deferredPromise';
+
+export type ReconnectOptions = {
+	auto: boolean;
+	delay: number;
+	maxAttempts: number;
+	onTimeout: boolean;
+};
+
+export interface WebSocketOptions {
+	protocol?: string;
+	headers?: OutgoingHttpHeaders;
+	/* eslint-disable @typescript-eslint/ban-types */
+	requestOptions?: Object;
+	clientConfig?: IClientConfig;
+	customTimeout: number;
```

Comment:
What's the difference between `customTimeout` and `timeout`?

Reply:
with new lib `isomorphic-ws` and `ws` its no longer required.